rust/src/action.rs

```diff
@@ -845,32 +844,87 @@ impl Action {
 
 /// Operation performed when creating a new log entry with one or more actions.
 /// This is a key element of the `CommitInfo` action.
-#[derive(Serialize, Deserialize, Debug)]
+#[allow(clippy::large_enum_variant)]
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
 pub enum DeltaOperation {
+    /// Represents a Delta `Create` operation.
+    /// Would usually only create the table, if also data is written,
+    /// a `Write` operations is more appropriate
+    Create {
+        /// The save mode used during the create.
+        mode: SaveMode,
+        /// The storage location of the new table
+        location: String,
+        /// The min reader and writer protocol versions of the table
+        protocol: Protocol,
+        /// Metadata associated with the new table
+        metadata: DeltaTableMetaData,
+    },
+
     /// Represents a Delta `Write` operation.
     /// Write operations will typically only include `Add` actions.
+    #[serde(rename_all = "camelCase")]
     Write {
         /// The save mode used during the write.
         mode: SaveMode,
         /// The columns the write is partitioned by.
-        partitionBy: Option<Vec<String>>,
+        partition_by: Option<Vec<String>>,
```

Comment:
We have that field here in case the table does not exist and we create it during the write operation. When it comes to changing the table schema and or partitioning I am still a bit fuzzy to be honest. I seem to remember reading something that changing partitioning in schema evolution is not allowed, but could not find a reference to it in the delta protocol definitions.

Reply:
> I seem to remember reading something that changing partitioning in schema evolution is not allowed
That sounds right to me.

I guess my point is I don't see the value in keeping it since it's recorded in metadata and shouldn't vary from that.

Reply:
We require that info when creating a new table during write. The original implementation also menations checking the coluumns against an exisiting table and failing otherwise. We do use it to crate a table, the check is not done right now.

Reply:
would repartition a valid use for this argument?